Description: Transition to Gradle 3.1
 This patch makes the build scripts compatible with Gradle 3.1
Author: Kai-Chung Yan (殷啟聰)
Last-Update: 2016-11-01
--- a/gradle/docs.gradle
+++ b/gradle/docs.gradle
@@ -40,10 +40,6 @@
 }
 
 def groovydocSpec = {
-    // hack to avoid errors when using gradle 3 to get the wrapper
-    // we won't ever build 2.4.x using a 3+ version - at least not
-    // without further changes
-    if (gradle.gradleVersion.startsWith('2.')) {
         use = true
         if (project != rootProject) source = project.sourceSets.main.allSource
         classpath = javadoc.classpath
@@ -51,7 +47,11 @@
         ext.doctitle = doc.title
         header = doc.title
         footer = doc.footer
-        overview = rootProject.file('src/main/overview.html')
+        if (gradle.gradleVersion.startsWith('2.')) {
+            overview = rootProject.file('src/main/overview.html')
+        } else {
+            overviewText = resources.text.fromFile(rootProject.file('src/main/overview.html'))
+        }
         includePrivate = false
         link 'http://docs.oracle.com/javaee/7/api/', 'javax.servlet.', 'javax.management.'
         link 'http://docs.oracle.com/javase/8/docs/api/', 'java.', 'org.xml.', 'javax.', 'org.w3c.'
@@ -59,7 +59,6 @@
         link 'http://junit.org/junit4/javadoc/latest/', 'org.junit.', 'junit.'
         link 'http://www.antlr2.org/javadoc/', 'antlr.'
         link 'http://commons.apache.org/proper/commons-cli/javadocs/api-release/', 'org.apache.commons.cli.'
-    }
 }
 
 allprojects {
